### PR TITLE
Fixed flappers

### PR DESCRIPTION
**Pull request by kozlovic**
Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/events_test.go [hunk 1]
```diff
@@ -813,6 +813,11 @@ func TestSystemAccountSystemConnectionLimitsHonored(t *testing.T) {
 		}
 		defer ncb1.Close()
 		tc++
+
+		// The account's connection count is exchanged between servers
+		// so that the local count on each server reflects the total count.
+		// Pause a bit to give a change for each server to process the update.
```
Comment (derekcollison) [inline server/events_test.go]: chance

File: server/server_test.go [hunk 2]
```diff
@@ -699,13 +699,22 @@ func TestLameDuckMode(t *testing.T) {
...
-	}
+	// Connections are saved in go routines, so although we have evaluated the number
+	// of connections in the server A to be 0, the polling of connection closed may
+	// need a bit more time.
+	checkFor(t, time.Second, 15*time.Millisecond, func() error {
+		cz := pollConz(t, srvA, 1, "", &ConnzOptions{State: ConnClosed})
+		if n := len(cz.Conns); n != total {
+			return fmt.Errorf("expected %v closed connections, got %v", total, n)
+		}
+		for _, c := range cz.Conns {
```
Comment (derekcollison) [inline server/server_test.go]: Probably put this outside checkFor loop.

File: test/new_routes_test.go [hunk 3]
```diff
@@ -1086,7 +1092,9 @@ func TestNewRouteStreamImport(t *testing.T) {
 	sendB("UNSUB 1\r\nPING\r\n")
 	expectB(pongRe)
 
-	checkExpectedSubs(0, srvA)
+	if err := checkExpectedSubs(0, srvA); err != nil {
```
Comment (derekcollison) [inline test/new_routes_test.go]: Did this not t.Fatalf on its on?

File: server/events_test.go (hunk 1 above)
Comment (kozlovic) [reply server/events_test.go]: also `for`->`to`: `give a chance to each server`.

File: server/server_test.go (hunk 2 above)
Comment (kozlovic) [reply server/server_test.go]: But we need `cz`. I could have the checkFor() that ensures that len(cz.Conns) is correct, then do another one outside of the checkFor().

File: test/new_routes_test.go (hunk 3 above)
Comment (kozlovic) [reply test/new_routes_test.go]: Not the one in the `/test` directory. The one in `/server` accepts `t` and so calls t.Fatalf(), but not this one. I could fix the 37 occurrences in `/test` package. You want me to do that?

Comment (derekcollison) [reply test/new_routes_test.go]: nah.. all good.

File: server/server_test.go (hunk 2 above)
Comment (kozlovic) [reply server/server_test.go]: Ok, I just added the poll/check for len in the checkFor() before the original code. I think it is better this way (use of checkReason() helper).

**Pull request merged**